root_draft_metadata.tf

```diff
@@ -46,9 +47,212 @@ module "draft_metadata_api_gateway" {
   }]
 }
 
+resource "aws_iam_role" "draft_metadata_api_gateway_execution_role" {
+  name = "TDRMetadataChecksAPIGatewayExecutionRole${title(local.environment)}"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Principal = {
+          Service = "apigateway.amazonaws.com"
+        },
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+
+  inline_policy {
+    name = "TDRMetadataChecksAPIGatewayStepFunctionExecutionPolicy${title(local.environment)}"
+    policy = jsonencode({
+      Version = "2012-10-17",
+      Statement = [
+        {
+          Effect   = "Allow",
+          Action   = "states:StartExecution",
+          Resource = module.draft_metadata_checks.step_function_arn
+        }
+      ]
+    })
+  }
+}
+
 module "draft_metadata_bucket" {
   source      = "./da-terraform-modules/s3"
   bucket_name = local.draft_metadata_s3_bucket_name
   common_tags = local.common_tags
   kms_key_arn = module.s3_internal_kms_key.kms_key_arn
 }
+
+data "aws_ssm_parameter" "backend_checks_keycloak_secret" {
```

Comment:
There is a story to create a new Keycloak client for the draft metadata, so this will need to be changed as part of that task: https://national-archives.atlassian.net/browse/TDRD-181

Reply:
:+1: